tests/src/core/testqgsrectangle.cpp

```diff
@@ -348,5 +349,20 @@ void TestQgsRectangle::dataStream()
   QCOMPARE( result, original );
 }
 
+void TestQgsRectangle::snappedToGrid()
+{
+  QgsRectangle original( 10.123, 20.333, 10.788, 20.788 );
+  QgsRectangle snapped = original.snappedToGrid( 0.1 );
+
+  QgsRectangle control( 10.1, 20.3, 10.8, 20.8 );
+
+  QVERIFY( qgsDoubleNear( snapped.xMinimum(), control.xMinimum(), 0.000001 ) );
```

Comment:
why not using `==` which already uses `qgsDoubleNear`

Reply:
1. I didn't know
2. The epsilon is a bit too small (`10.1` in control vs. the snapped `10.100000000000001`)